### PR TITLE
Linkage fhir

### PR DESCRIPTION
**Pull request by KennethSkylight**
This ticket moves the add_patient_identifier function into the fhir folder. It already takes in a FHIR bundle and handles the linking logic with generate_hash_str

### CODE REVIEW

**Review by spencer-skylight — CHANGES_REQUESTED** [review]
A few very minor changes to request, almost ready to go IMO.

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.formatting.provider": "black"
```
[the remark this answers is not in the document]
Comment (KennethSkylight) [reply .vscode/settings.json:2]: Removed

File: phdi/fhir/linkage/link.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import copy
+from phdi.fhir.utils import (
+    find_resource_by_type,
+    get_field,
+    get_one_line_address,
+)
+from ...linkage.link import generate_hash_str
```
[the remark this answers is not in the document]
Comment (KennethSkylight) [reply phdi/fhir/linkage/link.py:7]: Changed

File: phdi/fhir/linkage/link.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        able to reverse the hash into PII
+    :param overwrite: Whether to write the new standardizations
+        directly into the given bundle, changing the original data (True
+        is yes)
+    """
+    if not overwrite:
+        bundle = copy.deepcopy(bundle)
+
+    for resource in find_resource_by_type(bundle, "Patient"):
+        patient = resource.get("resource")
```
[the remark this answers is not in the document]
Comment (KennethSkylight) [reply phdi/fhir/linkage/link.py:31]: I see. So much of the contents of the for loop could be its own function that processes a single resource. 

A separate function can be used to loop through the bundle and reuse the single resource function. 

Is that right?

[the remark this answers is not in the document]
Comment (spencer-skylight) [reply phdi/fhir/linkage/link.py:31]: Yep, that's right.

[the remark this answers is not in the document]
Comment (KennethSkylight) [reply phdi/fhir/linkage/link.py:31]: I'm good with that

[the remark this answers is not in the document]
Comment (KennethSkylight) [reply phdi/fhir/linkage/link.py:31]: I think we should make it another ticket since this is just a port

[the remark this answers is not in the document]
Comment (spencer-skylight) [reply phdi/fhir/linkage/link.py:31]: Sounds good! I'm in clickup now doing planning, so I'll create a ticket for this now, and will approve the PR.

[the remark this answers is not in the document]
Comment (spencer-skylight) [reply phdi/fhir/linkage/link.py:31]: https://app.clickup.com/t/3c0axqb

**Review by spencer-skylight — APPROVED** [review]
LGTM!

**Pull request merged**